handlers/product-switch-api/src/contributionToSupporterPlus.ts

```diff
@@ -56,56 +65,72 @@ export const switchToSupporterPlus = async (
 	};
 };
 
-const getContributionRefundAmount = (
-	zuoraResponse: ZuoraPreviewResponse,
+export const handleMissingRefundAmount = (
 	catalogInformation: CatalogInformation,
-	termStartDate: Date,
+	subscription: ZuoraSubscription,
+	currentDate: Date,
 ): number => {
-	const termStartDateDayOfMonth = termStartDate.getUTCDate();
-	const dayOfMonthToday = new Date().getUTCDate();
+	const ratePlan = getIfDefined(
+		subscription.ratePlans.find(
+			(ratePlan: RatePlan) =>
+				ratePlan.productRatePlanId ===
+				catalogInformation.contribution.productRatePlanId,
+		),
+		'No matching RatePlan found in Subscription,',
+	);
```

Comment:
not one for this PR, but this is a common pattern and it needs so many lines of boilerplate.  It would be good to invent something where we just wrap a subscription and say something like
```suggestion
	const ratePlan = getRatePlanById(
		subscription,
		catalogInformation.contribution,
	);
```
or
```suggestion
	const ratePlan = getContributionRatePlan(subscription);
```
but I realise that's my scala brain at work